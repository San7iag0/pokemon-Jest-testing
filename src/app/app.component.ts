import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})

export class AppComponent {
  
  constructor(){}

  onInit() {
    this.sumar(2, 2);
  }

  title = 'angular-testing';

  sumar(a: number, b: number){
    a + b
  }
}
