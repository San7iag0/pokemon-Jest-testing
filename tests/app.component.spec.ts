import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {

  beforeEach(async () => {

    let appComponent: AppComponent;

    await TestBed.configureTestingModule({
      providers: [AppComponent],
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
    appComponent = TestBed.inject(AppComponent);

  });


  test('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  test(`should have as title 'angular-testing'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component.title).toEqual('angular-testing');
  });

  test('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    
    const h1 = compiled.querySelector('h1'); // look for the h1
    expect( h1?.textContent ).toContain( component.title );
  });

  test('should test if changes on html', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges(); // detect changes 
    const compiled = fixture.nativeElement as HTMLElement;
    expect( compiled ).toMatchSnapshot();

  })

  it('adds  variables', () => {
    const fixture = TestBed.inject(AppComponent);
    jest.spyOn(fixture, 'sumar')
    expect(fixture.sumar).toBeCalled
  })
});
