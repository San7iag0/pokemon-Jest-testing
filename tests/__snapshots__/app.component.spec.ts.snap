// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent debe hacer match con el snapshot 1`] = `
<div
  id="root3"
>
  <h1>
    angular-testing
  </h1>
</div>
`;

exports[`AppComponent should test if changes on html 1`] = `
<div
  id="root3"
>
  <h1>
    angular-testing
  </h1>
</div>
`;
